packages/web-client/app/components/common/degraded-service-banner/index.ts

```diff
@@ -0,0 +1,71 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import * as Sentry from '@sentry/browser';
+import { task, TaskGenerator, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import config from '@cardstack/web-client/config/environment';
+
+export default class DegradedServiceBannerComponent extends Component {
+  @tracked notificationShown: boolean = false;
+  @tracked notificationBody: string | null = null;
+  @tracked impact: 'none' | 'minor' | 'major' | 'critical' | null = null;
+
+  statusPageUrl = config.urls.statusPageUrl;
+
+  get isSevere() {
+    return this.impact === 'major' || this.impact === 'critical';
+  }
+
+  @action startPolling() {
+    taskFor(this.pollForStatusTask).perform();
+  }
+
+  @action stopPolling() {
+    taskFor(this.pollForStatusTask).cancelAll();
+  }
+
+  @task *pollForStatusTask(): TaskGenerator<void> {
+    while (true) {
+      let statusData = yield this.getDegradationStatusData();
+
+      if (statusData) {
+        this.notificationShown = true;
+        this.impact = statusData.impact;
+        this.notificationBody = statusData.name;
+      } else {
+        this.notificationShown = false;
+        this.notificationBody = null;
+        this.impact = null;
+      }
+
+      yield timeout(1000 * 60);
+    }
+  }
```

Comment:
There might be 2 of these banners in the design because we display one in a workflow, so this might be better as a service?

Reply:
Sounds good, will make a service